### PR TITLE
Download files using zak token (admin authentication)

### PR DESCRIPTION
**Pull request by nataliemona**

### CODE REVIEW

**Comment by nataliemona** [comment]
* Changed the downloader to use a zak token (admin authentication) to download files
* Set up the lambda code to be put in a folder within a more general S3 bucket (our general lambda code bucket for all projects)
* Support multiple schedule times for one zoom meeting series id
* Add an example of the format for the schedule .json file

File: functions/zoom-downloader.py [hunk 1]
```diff
@@ -284,59 +280,39 @@ def stream_file_to_s3(file, uuid, track_sequence):
...
+    # use zak token to get download stream
+    logger.info("requesting {}".format(download_url))
 
-def get_connection_using_play_url(play_url):
-    file_url = retrieve_url_from_play_page(play_url)
+    r = requests.get("{}?zak={}".format(download_url, admin_token), allow_redirects=False)
+    if 'Location' in r.headers:
+        location = r.headers['Location']
+        zoom_name = location.split('?')[0].split('/')[-1]
+    else:
```
Comment (lbjay) [inline functions/zoom-downloader.py:294]: so the zoom behavior is not consistent here?

File: functions/zoom-downloader.py [hunk 2]
```diff
@@ -284,59 +280,39 @@ def stream_file_to_s3(file, uuid, track_sequence):
...
-    try:
-        zoom_name = r.headers['Content-Disposition'].split("=")[-1]
-        logger.info("got filename {}".format(zoom_name))
-    except KeyError:
-        raise PermanentDownloadError("File name not in headers: {}".format(download_url))
-
-    return r, zoom_name
+def get_admin_token():
+    # get admin user id from admin email
+    r = requests.get("https://api.zoom.us/v2/users/{}".format(ZOOM_ADMIN_EMAIL),
```
Comment (lbjay) [inline functions/zoom-downloader.py]: we can't use `ZOOM_API_BASE_URL` here?

File: tests/test_downloader.py [hunk 3]
```diff
@@ -157,23 +157,4 @@ def side_effect(*args, **kwargs):
...
-                    ('<div id=password_form> </div>', 'Password protected')]
-
-        for content, message in rec_data:
-            with pytest.raises(downloader.PermanentDownloadError) as excinfo:
-                req_mock.get(requests_mock.ANY, status_code=200,
-                             content=content.encode('utf-8'))
-                downloader.retrieve_url_from_play_page("https://www.example.com")
-            assert(message in str(excinfo.value))
\ No newline at end of file
+        downloader.get_recording_data(call[0])
```
Comment (lbjay) [inline tests/test_downloader.py:160]: mostly aesthetic, but you should see if your editor has an option to automatically insert a newline at the end of every file

**Review by lbjay — APPROVED** [review]

File: functions/zoom-downloader.py (hunk 1 above)
Comment (nataliemona) [reply functions/zoom-downloader.py:294]: File name is in 'Content Disposition' for .vtt (transcript files) but 'Location' for .mp4, .m4a, and .txt.

**Pull request merged**